module/input/SensorFilter/src/MotionModel.hpp

```diff
@@ -206,7 +207,7 @@ namespace module::input {
 
         Eigen::Matrix<Scalar, 4, 1> predict(const Eigen::Matrix<Scalar, size, 1>& state,
                                             const MeasurementType::FLAT_FOOT_ORIENTATION&) {
-            return StateVec(state).Rwt.coeffs();
+            return Eigen::Matrix<Scalar, 4, 1>(state.template segment<4>(StateVec::QX));
```

Comment:
Can't you just do this?
```suggestion
            return state.template segment<4>(StateVec::QX);
```

Reply:
Yep. Good eye.